src/compiler/checker.ts

```diff
@@ -9656,16 +9656,22 @@ namespace ts {
         function checkAssertion(node: AssertionExpression) {
             const exprType = getRegularTypeOfObjectLiteral(checkExpression(node.expression));
             const targetType = getTypeFromTypeNode(node.type);
-            if (produceDiagnostics && targetType !== unknownType) {
-                const widenedType = getWidenedType(exprType);
+            if (produceDiagnostics) {
+                if (node.type.kind === SyntaxKind.TypePredicate) {
+                    error(node.type, Diagnostics.A_type_predicate_is_not_allowed_as_part_of_a_type_assertion_expression);

```

Comment:
Rather than creating a new error message, why not call `checkTypePredicate`, which aught to error here?